xcode/Subconscious/Shared/Services/UserProfileService.swift

```diff
@@ -345,60 +375,56 @@ actor UserProfileService {
         }
     }
     
-    func loadOurProfileFromMemo() async throws -> UserProfile {
-        let did = try await noosphere.identity()
-        let cid = try await noosphere.version()
-        
-        return try await self.loadProfileFromMemo(
-            did: did,
-            address: Slashlink.ourProfile,
-            resolutionStatus: .resolved(cid)
-        )
-    }
+//    func loadOurProfileFromMemo() async throws -> UserProfile {
+//        let did = try await noosphere.identity()
+//        let cid = try await noosphere.version()
+//
+//        return try await self.loadProfileFromMemo(
+//            did: did,
+//            address: Slashlink.ourProfile,
+//            resolutionStatus: .resolved(cid)
+//        )
+//    }
     
-    nonisolated func loadOurProfileFromMemoPublisher(
-    ) -> AnyPublisher<UserProfile, Error> {
-        Future.detached {
-            try await self.loadOurProfileFromMemo()
-        }
-        .eraseToAnyPublisher()
-    }
+//    nonisolated func loadOurProfileFromMemoPublisher(
+//    ) -> AnyPublisher<UserProfile, Error> {
+//        Future.detached {
+//            try await self.loadOurProfileFromMemo()
+//        }
+//        .eraseToAnyPublisher()
+//    }
     
+    /// Produces a `UserProfile` for the passed address.
+    /// If a `did` is provided we can skip resolution and return faster.
     func buildUserProfile(
-        address: Slashlink
+        address: Slashlink,
+        did: Did?
     ) async throws -> UserProfile {
         let version = try await self.noosphere.version()
+        let cacheKey = address.isOurs ? Slashlink.ourProfile : address
         
-        if let petname = address.petname,
-           let cacheHit = self.cache[petname],
+        if let cacheHit = self.cache[cacheKey],
            cacheHit.version == version {
             return cacheHit.profile
         }
         
-        let did = try await self.noosphere.resolve(peer: address.peer)
-        
-        let resolutionStatus: ResolutionStatus = try await Func.run {
-            guard let petname = address.petname else {
-                let sphere = try await self.noosphere.sphere(address: address)
-                let cid = try await sphere.version()
-                return .resolved(cid)
-            }
-            
-            return try await self.addressBook.resolutionStatus(petname: petname)
+        var identity = did
+        if identity == nil {
+            identity = try await self.noosphere.identity()
```

Comment:
doesn't make a lot of sense to me, I think we want resolve here, but this code might go away anyway